src/webgpu/shader/validation/shader_io/builtins.spec.ts

```diff
@@ -133,3 +133,34 @@ g.test('type')
     );
     t.expectCompileResult(expectation, code);
   });
+
+g.test('nesting')
+  .desc(`Test validation of nested built-in variables`)
+  .params(u =>
+    u
+      .combine('target_stage', ['fragment', ''] as const)
+      .combine('target_io', ['in', 'out'] as const)
+      .beginSubcases()
+  )
+  .fn(t => {
+    let code = '';
+
+    // Generate a struct that contains a sample_mask builtin, nested inside another struct.
+    code += 'struct Inner {\n';
```

Comment:
backticks are multi-line, so why the concatenation?

Reply:
Done.